packages/server-wallet/src/models/channel.ts

```diff
@@ -414,8 +418,71 @@ export class Channel extends Model implements RequiredColumns {
     return !!this.supported && this.supported.turnNum >= 2 * this.nParticipants - 1;
   }
 
-  public get isDirectFunded(): boolean {
-    return this.protocolState.directFundingStatus === 'Funded';
+  // Does the channel have funds to pay out to all allocation items?
+  public get isFullyDirectFunded(): boolean {
+    return this.isDirectFunded(true);
+  }
+
+  // Does the channel have any funds that I can claim?
+  public get isPartlyDirectFunded(): boolean {
+    return this.isDirectFunded();
+  }
+
+  private isDirectFunded(fully?: boolean): boolean {
+    const outcome = this.supported?.outcome;
+    if (!outcome) {
+      throw new Error(`Channel passed to isFullyDirectFunded has no supported state`);
+    }
+
+    const {assetHolderAddress, allocationItems} = checkThat(outcome, isSimpleAllocation);
+    const channelFunds = this.funding.find(f => f.assetHolder === assetHolderAddress);
+    if (!channelFunds) return false;
+
+    if (fully) {
+      const fullFunding = allocationItems.map(a => a.amount).reduce(BN.add, BN.from(0));
+      return BN.gte(channelFunds.amount, fullFunding);
+    } else {
+      const fundingBeforeMe = this.fundingMilestones[0];
+      return BN.gt(channelFunds.amount, fundingBeforeMe);
+    }
+  }
+
+  /**
+   * Returns a triple of balance [targetBefore, targetAfter, targetTotal], where
+   *  - targetBefore is the balance where I should start depositing
+   *  - targetAfter should be the balance where I stop depositing
+   *  - targetTotal is the total amount that should be in the channel
+   *
+   * @param channel
+   */
+  public get fundingMilestones(): [Uint256, Uint256, Uint256] {
```

Comment:
Perhaps this is my preference but I'd prefer returning a type like `{ targetBefore: Uint256, targetAfter: Uint256, targetTotal: Uint256}`. I feel like I'd often get the correct order of the array confused (although to be fair, it will display in intellisense).

Reply:
This is a good point! `fundingMilestones` was copy/pasted from another location, so I didn't think too deeply about the interface. There are 3 PRs that are based on this branch. So my preference is to file an issue and address the issue shortly after this PR is merged.

FIled https://github.com/statechannels/statechannels/issues/3268.